### ISSUE TITLE
Has the corrected reverse KL not been implemented yet? I don't see the calculation for the formula  (log_prob_actor - log_prob_ref) - (log_prob_actor_base - log_prob_ref_base) in the code of dp_actor.py

### ISSUE THREAD

**Issue by QWE-CXZ** [RUCBM/G-OPD#7]

**Comment by keven980716** [comment]
Hi!We have implemented the reward correction. Just set +actor_rollout_ref.model.base_model_path=teacher's pre-RL model in the script~

**Comment by QWE-CXZ** [comment]
Thank you for your reply! However, it seems the implementation in dp_actor.py doesn't match the formula mentioned above. Is this logic implemented elsewhere in the codebase?
### The code at dp_actor.py
```if self.config.policy_loss.only_reverse_kl_advantages:
                        # Corrected reverse KL with base model normalization if base log probs are available
                        # Formula: (log_prob_actor - log_prob_ref) - (log_prob_actor_base - log_prob_ref_base)
                        # This removes the base model bias from both actor and ref models
                        if "base_log_prob" in model_inputs and "base_ref_log_prob" in model_inputs:
                            lambda_vals = self.config.policy_loss.lambda_vals

                            if self.config.policy_loss.multi_teacher_distill:
                                #### multi-teacher distillation ####
                                if "opd_teacher" in model_inputs:
                                    opd_teacher = model_inputs["opd_teacher"]
                                    batch_size = old_log_prob.shape[0]

                                    reverse_kl = torch.zeros_like(old_log_prob)

                                    for i in range(batch_size):
                                        teacher_type = opd_teacher[i] if isinstance(opd_teacher, (list, tuple)) else opd_teacher
                                        # TODO: need to improve the logic here
                                        if teacher_type == "math":
                                            if lambda_vals == 1.0:
                                                reverse_kl[i] = old_log_prob[i] - model_inputs["ref_log_prob"][i]
                                            else:
                                                reverse_kl[i] = old_log_prob[i] - model_inputs["base_log_prob"][i] - (model_inputs["ref_log_prob"][i] - model_inputs["base_log_prob"][i]) * lambda_vals
                                        elif teacher_type == "code":
                                            if lambda_vals == 1.0:
                                                reverse_kl[i] = old_log_prob[i] - model_inputs["base_ref_log_prob"][i]
                                            else:
                                                reverse_kl[i] = old_log_prob[i] - model_inputs["base_log_prob"][i] - (model_inputs["base_ref_log_prob"][i] - model_inputs["base_log_prob"][i]) * lambda_vals
                                        else:
                                            reverse_kl[i] = old_log_prob[i] - model_inputs["ref_log_prob"][i]
                                else:
                                    reverse_kl = old_log_prob - model_inputs["ref_log_prob"]
                                #### multi-teacher distillation ####
                            else:
                                #### single-teacher distillation ####
                                reverse_kl = old_log_prob - model_inputs["base_log_prob"]
                                reward_correction = model_inputs["ref_log_prob"] - model_inputs["base_log_prob"]

                                if lambda_vals == 1.0:
                                    reverse_kl = old_log_prob - model_inputs["ref_log_prob"]
                                else:
                                    reverse_kl = reverse_kl - reward_correction * lambda_vals
                                #### single-teacher distillation ####
                        else:
                            # Standard reverse KL: log(π_actor / π_ref) = log_prob_actor - log_prob_ref
                            reverse_kl = old_log_prob - model_inputs["ref_log_prob"]
                        advantages = (- (reverse_kl))```

**Comment by keven980716** [comment]
In G-OPD (assume in the single-teacher distillation setting), the token-level advantage is -[ log_prob_actor (old_log_prob in the code) - log_prob_teacher (model_inputs["ref_log_prob"] in the code) - (lambda - 1) (log_prob_ref (model_inputs["base_log_prob"] in the code) - log_prob_teacher) ], so there are totally 3 models here, not 4.

**Comment by QWE-CXZ** [comment]
Got it! Thank you!
